api/registry.go

```diff
@@ -131,6 +132,11 @@ func (h *RegistryHandlers) registryDatasetHandler(w http.ResponseWriter, r *http
 
 	err = h.RegistryRequests.GetDataset(&ref, res)
 	if err != nil {
+		// If error was that the dataset wasn't found, send back 404 Not Found.
+		if strings.HasPrefix(err.Error(), "error 404") {
```

Comment:
we should make this a more robust thing later, like, you know, returning proper status codes :/